packages/tiled/src/TilesetLoader.js

```diff
@@ -0,0 +1,53 @@
+import url from 'url';
+
+import { LoaderResource } from '@pixi/loaders';
+
+/**
+ * {@link PIXI.Loader Loader} middleware for loading TSX (Tile Set XML) files.
+ * This automatically generate a Texture resource.
+ * @class
+ * @memberof PIXI
+ * @extends PIXI.Loader~LoaderPlugin
+ */
+export default class TilesetLoader
+{
+    /**
+     * Called after a resource is loaded.
+     * @see PIXI.Loader~loaderMiddleware
+     * @param {PIXI.LoaderResource} resource
+     * @param {function} next
+     */
+    static use(resource, next)
+    {
+        const tileset = resource.data;
+        const imageResourceName = `${resource.name}_image`;
+
+        // skip if no data, not a document, it isn't a tileset, contains no image, or the image resource already exists
+        if (!tileset
+            || resource.type !== LoaderResource.TYPE.XML
+            || resource.extension !== 'tsx'
+            || !tileset.getElementsByTagName('image').length
+            || this.resources[imageResourceName]
+        )
+        {
+            next();
+
+            return;
+        }
+
+        const loadOptions = {
+            crossOrigin: resource.crossOrigin,
+            parentResource: resource,
+        };
+
+        const imageElement = tileset.getElementsByTagName('image')[0];
```

Comment:
In @englercj's loader middleware example for loading tiled JSON files, he is handling many more image instances. In this case, the loader plugin only loads the first image element within the tileset. I'm not super familiar with the format, but shouldn't this handle more images?

Reply:
> he is handling many more image instances

@bigtimebuddy I think that is due to the fact that @englercj is loading the [JSON Map Format] which represents a complete map. It is the JSON counterpart of the [TMX Map Format].

The plugin that I have written here is only for [TSX (Tile Set XML) files] which are external tilesets and may be referenced by either [JSON Map Format] or [TMX Map Format].

I'm planning to also write plugins for the map formats (both TMX/XML and JSON) but thought that I'd start with a smaller contribution that can be finished on a weekend. :smiley: 

>  shouldn't this handle more images?

Yes, I was misreading the format. I only thought of

```xml
<?xml version="1.0" encoding="UTF-8"?>
<tileset name="building1" tilewidth="16" tileheight="16" tilecount="64" columns="8">
 <image source="building1.png" width="128" height="128"/>
</tileset>
```

but this is also possible:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<tileset name="building1" tilewidth="209" tileheight="144" tilecount="2" columns="13">
 <grid orientation="orthogonal" width="16" height="16"/>
 <tile id="0">
  <image width="128" height="128" source="building1.png"/>
 </tile>
 <tile id="1">
  <image width="209" height="144" source="paint-tanuki.png"/>
 </tile>
</tileset>
```

Thank you for making me check this! :+1: I will adjust the loader plugin and make sure it is covered by the tests.

[JSON Map Format]: http://doc.mapeditor.org/en/latest/reference/json-map-format/
[TMX Map Format]: http://doc.mapeditor.org/en/latest/reference/tmx-map-format/
[TSX (Tile Set XML) files]: http://doc.mapeditor.org/en/latest/reference/tmx-map-format/#tileset

Reply:
Just FYI, there are two types of tilesets for tilelayers. One that has a single image of tiles (you're first xml block), and one that is a "multi-image tileset" where each individual tile is a single image (you're second xml block).